# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/Barrier.h
# Commit: e9c2f256240a
# Full Hash: e9c2f256240a1f9bb09b10ac641bbd141eedb809
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 2 - Remove owned BaseShapes. r=jonco
#   
#   At this point all BaseShapes are unowned (= immutable, shared) so we can remove the
#   UnownedBaseShape type and the `BaseShape::unowned_` field.
#   
# ==============================================================================

diff -r 1ae4c30bb931 -r e9c2f256240a js/src/gc/Barrier.h
--- a/js/src/gc/Barrier.h	Sat Mar 06 01:05:19 2021 +0000
+++ b/js/src/gc/Barrier.h	Sat Mar 06 01:05:19 2021 +0000
@@ -1115,7 +1115,6 @@
 class ScriptSourceObject;
 class Shape;
 class BaseShape;
-class UnownedBaseShape;
 class WasmInstanceObject;
 class WasmTableObject;
 
@@ -1137,7 +1136,6 @@
 using GCPtrScript = GCPtr<JSScript*>;
 using GCPtrString = GCPtr<JSString*>;
 using GCPtrShape = GCPtr<Shape*>;
-using GCPtrUnownedBaseShape = GCPtr<UnownedBaseShape*>;
 using GCPtrObjectGroup = GCPtr<ObjectGroup*>;
 using GCPtrValue = GCPtr<Value>;
 using GCPtrId = GCPtr<jsid>;